### PR TITLE
Adds new log viewer component across all programs in CDAP

### PR DESCRIPTION
**Pull request by PGuimarais**
- Fixes log viewer to handle use cases based on the status of the program + logs availability.
- Adds new log viewer to all programs across CDAP (workflows, workers, sparks, mapreduce, services)
- Minor styling cleanups.

Bamboo build: http://builds.cask.co/browse/CDAP-DRC4099-1

### CODE REVIEW

**Comment by elfenheart** [comment]
admins?

**Comment by PGuimarais** [comment]
@elfenheart Lets take admin log viewer in a new PR. The current new log viewer doesn't account for `/system/services/*` end point and the existing `admin-log-viewer` is not really the `old-log-viewer` its a separate log viewer that looks like old-log-viewer 😄

File: cdap-ui/app/directives/log-viewer/log-viewer.html [hunk 1]
```diff
@@ -38,7 +38,12 @@
 
     <div class="log-actions">
       <div class="log-action-container">
-        <a  href="{{LogViewer.rawUrl}}" target="_blank"><button class="log-action-btn view-raw">View Raw</button></a>
+        <a  href="{{LogViewer.rawUrl}}"
+            target="_blank">
+          <button class="log-action-btn view-raw" ng-disabled="!LogViewer.rawUrl.length">
```
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.html]: you should be able to style the anchor tag directly without nesting a button inside

File: cdap-ui/app/directives/log-viewer/log-viewer.js [hunk 2]
```diff
@@ -316,6 +328,7 @@ function LogViewerController ($scope, LogViewerStore, myLogsApi, LOGVIEWERSTORE_
         }
       },
       (err) => {
+        this.setDefault();
```
Comment (elfenheart) [inline cdap-ui/app/directives/log-viewer/log-viewer.js:331]: set loading to false on error

**Comment by sreevatsanraman** [comment]
LGTM. Address any follow up comments in a new PR against release/3.5

File: cdap-ui/app/directives/log-viewer/log-viewer.js (hunk 2 above)
Comment (ajainarayanan) [inline cdap-ui/app/directives/log-viewer/log-viewer.js:331]: Isn't setDefault() already setting it to false?

**Pull request merged**